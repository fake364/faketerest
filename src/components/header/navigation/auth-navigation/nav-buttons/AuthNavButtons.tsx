import React from 'react';
import Tooltip from '../../../../../common/components/tooltip/Tooltip';
import CircleIconButton from '../../../../../common/components/buttons/CircleIconButton';
import { FaBell } from '@react-icons/all-files/fa/FaBell';
import { FaCommentDots } from '@react-icons/all-files/fa/FaCommentDots';
import UserRoundButton from './user-button/UserRoundButton';
import useTranslation from 'next-translate/useTranslation';
import DropdownRootElement from '../../../../../common/components/buttons/buttonDropdown/DropdownRootElement';
import { useSelector } from 'react-redux';
import { RootState } from '../../../../../redux/types';
import { NotificationType } from '../../../../../common/types/events/EventTypes';
import style from './AuthNavButtons.module.css';
import NotificationWrapper from './notificationWrapper/NotificationWrapper';
import NotificationDropdown from './notificationDropdown/NotificationDropdown';

type Props = {};

const AuthNavButtons: React.FC<Props> = () => {
  const { t } = useTranslation('common');
  const notifications: NotificationType[] = useSelector(
    (state: RootState) => state.metadata.notifications
  );
  console.log('lelell', notifications);
  const unreadNotifications = notifications.filter(
    ({ payload: { hasBeenRead } }) => !hasBeenRead
  );
  return (
    <>
      <div className="relative">
        {unreadNotifications.length > 0 && (
          <div className={style.notificationBubble}>
            {unreadNotifications.length}
          </div>
        )}
        <NotificationDropdown />
      </div>
      <Tooltip text={t('header.tooltips.messages')}>
        <CircleIconButton
          Icon={FaCommentDots}
          className="px-[12px]"
          onClick={null}
        />
      </Tooltip>
      <UserRoundButton />
    </>
  );
};

export default AuthNavButtons;
