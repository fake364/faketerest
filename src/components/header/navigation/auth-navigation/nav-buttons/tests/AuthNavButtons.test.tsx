import React from 'react';
import { shallow } from 'enzyme';
import AuthNavButtons from '../AuthNavButtons';

describe('AuthNavButtons', () => {
  it('should render some buttons with tooltips', () => {
    const wrapper = shallow(<AuthNavButtons />);
    expect(wrapper).toMatchSnapshot();
  });
});
