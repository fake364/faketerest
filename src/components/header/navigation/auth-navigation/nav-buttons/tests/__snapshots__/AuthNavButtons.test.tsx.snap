// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNavButtons should render some buttons with tooltips 1`] = `
<Fragment>
  <Tooltip
    text="Уведомления"
  >
    <CircleIconButton
      Icon={[Function]}
      className="px-[12px]"
      onClick={null}
    />
  </Tooltip>
  <Tooltip
    text="Сообщения"
  >
    <CircleIconButton
      Icon={[Function]}
      className="px-[12px]"
      onClick={null}
    />
  </Tooltip>
  <UserRoundButton />
</Fragment>
`;
